From 9cd358d29764a588d825b52a106460a732a2e819 Mon Sep 17 00:00:00 2001
From: "zongren.yin" <zongren.yin@amlogic.com>
Date: Thu, 3 Aug 2023 15:55:40 +0800
Subject: [PATCH 12/15] hdmi: fix hdmi audio crash [1/1]

PD#SWPL-132030

Problem:
Hdmi tx audio not support 32bit sample size.

Solution:
Set cs_sw_len as 0xf while 32bit audio sample size.

Fixed conflicts:
NA

Verify:
local A311D build

Signed-off-by: zongren.yin <zongren.yin@amlogic.com>
Change-Id: I20f6c81c06683c2b9bff626d1d1b6a1f2afa0273
---
 drivers/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_audio.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_audio.c b/drivers/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_audio.c
index 5e936b8eb..3ebeaee31 100644
--- a/drivers/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_audio.c
+++ b/drivers/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_audio.c
@@ -39,7 +39,8 @@ static const unsigned char cs_sw_len[] = {
 	0x0,
 	0x2, /*16 bits*/
 	0x3, /*20 bits*/
-	0xb  /*24 bits*/
+	0xb, /*24 bits*/
+	0xf  /*32 bits*/
 };
 
 static void
-- 
2.25.1

